src/db.rs

```diff
@@ -12,7 +13,34 @@ const CACHE_EXPIRE: Duration = Duration::from_secs(90 * 60);
 #[derive(Debug, Serialize, Deserialize)]
 pub struct CacheEntry {
     pub from: SystemTime,
-    pub found: bool,
+    pub crate_status: CrateStatus,
+}
+
+/// The current status of a crate in Debian.
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub enum CrateStatus {
+    Available,
+    AvailableInNew,
+    Outdated,
```

Comment:
I think Outdated here is a bit ambiguous, isn't there two cases here: Too new version in Debian, and Too old version in Debian?

Reply:
Can't that be resolved like this:
```rust
pub enum CrateStatus [
pub enum CrateStatus {
    Available,
    AvailableInNew,
    Outdated_Needs_Update,
    Newer_Needs_Patching,
    Missing,
}
```
and so on for the rest of the code. That's a feature I really'd like to see.